File: CodingTest/11thStreet/200929/1.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(string & s) {
    int count1 = 0, count2 = 0, s_len = s.length(), answer = 0;
    for (int i = 0; i < s_len; ++i) {
        if (s[i] == 'a') {
            count1++;
        }
        else {
            count2++;
            count1 = 0;
        }
        if (count1 == 3) {
            return -1;
        }
    }
    answer = 2 * (count2 + 1) - (s_len - count2);
    
    return answer;
}

File: CodingTest/NHN/201127/1.java
//핵심 소스코드의 설명을 주석으로 작성하면 평가에 큰 도움이 됩니다.
public class Solution{
    public long solution(int n){
        long answer = 0;
        long[] fibo = new long[n+1];
        fibo[0] = (long) 0;
        
        if (n == 1) {
            answer = 1;
        }
        
        if (n > 1) {
            fibo[1] = 1;
            for (int i = 2; i <= n; i++) {
                fibo[i] = fibo[i-1] + fibo[i-2];
            }
        
            answer = fibo[n];
        }
        
        
        return answer;
    }
}